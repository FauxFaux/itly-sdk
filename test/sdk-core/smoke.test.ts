/* eslint-disable import/no-unresolved, global-require, import/extensions */
import itly from '../../packages/sdk-core/lib';
import CustomPlugin from '../../test/src/CustomPlugin';

// const testSchemas = require('./schemas/basic-schema.json');
// new SchemaValidator(testSchemas),

test('should load and track events to a custom destination (no validation)', () => {
  const spy = jest.spyOn(console, 'log');

  const userId = 'test-user-id';

  // new SchemaValidator(testSchemas)
  itly.load({
    environment: 'production',
    context: {
      requiredString: 'A required string',
      optionalEnum: 'Value 1',
    },
    plugins: [new CustomPlugin()],
  });

  itly.track(userId, {
    name: 'Event No Properties',
  });

  itly.track(userId, {
    name: 'Event With All Properties',
    properties: {
      requiredString: 'A required string',
      requiredNumber: 2.0,
      requiredInteger: 42,
      requiredEnum: 'Enum1',
      requiredBoolean: false,
      requiredConst: 'some-const-value',
      requiredArray: ['required', 'array'],
      optionalString: 'I\'m optional!',
    },
  });

  try {
    itly.track(userId, {
      name: 'EventMaxIntForTest',
      properties: {
        intMax10: 20,
      },
    });
  } catch (e) {
    // do nothing
  }

  const customOnly = itly.getPlugin('custom');
  // eslint-disable-next-line no-console
  console.log('CustomPlugin.id()', customOnly!.id());

  expect(spy.mock.calls).toMatchSnapshot();

  spy.mockRestore();
});
