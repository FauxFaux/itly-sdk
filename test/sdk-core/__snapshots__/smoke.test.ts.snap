// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load and track events to a custom destination (no validation) 1`] = `
Array [
  Array [
    "CustomPlugin: ",
    "track() userId='test-user-id' event='Event No Properties' properties=undefined",
  ],
  Array [
    "CustomPlugin: ",
    "track() userId='test-user-id' event='Event With All Properties' properties={\\"requiredString\\":\\"A required string\\",\\"requiredNumber\\":2,\\"requiredInteger\\":42,\\"requiredEnum\\":\\"Enum1\\",\\"requiredBoolean\\":false,\\"requiredConst\\":\\"some-const-value\\",\\"requiredArray\\":[\\"required\\",\\"array\\"],\\"optionalString\\":\\"I'm optional!\\"}",
  ],
  Array [
    "CustomPlugin: ",
    "track() userId='test-user-id' event='EventMaxIntForTest' properties={\\"intMax10\\":20}",
  ],
  Array [
    "CustomPlugin.id()",
    "custom",
  ],
]
`;
